nav2_behaviors/plugins/drive_on_heading.cpp

```diff
@@ -17,27 +17,31 @@
 #include <memory>
 #include <utility>
 
-#include "back_up.hpp"
+#include "drive_on_heading.hpp"
 #include "nav2_util/node_utils.hpp"
+#include "nav2_msgs/action/back_up.hpp"
 
 using namespace std::chrono_literals;
 
 namespace nav2_behaviors
 {
 
-BackUp::BackUp()
-: TimedBehavior<BackUpAction>(),
-  feedback_(std::make_shared<BackUpAction::Feedback>())
+template<typename ActionT>
+DriveOnHeading<ActionT>::DriveOnHeading()
+: TimedBehavior<ActionT>(),
+  feedback_(std::make_shared<typename ActionT::Feedback>())
 {
 }
 
-BackUp::~BackUp()
+template<typename ActionT>
+DriveOnHeading<ActionT>::~DriveOnHeading()
 {
 }
 
-void BackUp::onConfigure()
+template<typename ActionT>
+void DriveOnHeading<ActionT>::onConfigure()
 {
-  auto node = node_.lock();
+  auto node = this->node_.lock();
```

Comment:
Why the moves to `this`'s in a few places?

Reply:
https://isocpp.org/wiki/faq/templates
Look at the answer to the question
Why am I getting errors when my template-derived-class uses a member it inherits from its template-base-class?

Reply:
OK - I do like that the best of the options :+1: